### PR TITLE
[SWE-16] Login Page Backend

### PR DESCRIPTION
**Pull request by sammyoto**
INCOMPLETE. DO NOT PULL. Implemented a backend authentication service with cognito similar to what @datderek implemented. Can't yet login with my user sammyoto. Keep getting a user not confirmed error. 

    error signing in UserNotConfirmedException: { code: 'UserNotConfirmedException'}

Communication with backend works. Seems to be an issue with cognito sending out confirmation emails to confirm users.

### CODE REVIEW

**Comment by sammyoto** [comment]
I also have significantly more files added to the amplify folder than @datderek. Please help @jackfales.

File: amplify/team-provider-info.json
Comment (datderek) [inline amplify/team-provider-info.json:1]: Not too sure about this file specifically, it should be shared amongst team members working on the project. However this [article](https://docs.amplify.aws/cli/teams/shared/#sharing-projects-within-the-team) also mentions how if we plan on sharing our project publicly then this should be removed. See bottom of article.

File: package.json
Comment (datderek) [inline package.json:12]: I believe the Amplify UI library just adds premade UI components that are already set up to interface with their services. I omitted this since we have our own UIs set up already.

We haven't used them yet, so maybe just remove it for now? Thoughts @jackfales?

File: pages/api/login.js
Comment (datderek) [inline pages/api/login.js:13]: Need to send HTML response indicating success or failure (and the error message to display on failure).

File: pages/login.js
Comment (datderek) [inline pages/login.js:10]: Remove extra line

Comment (datderek) [inline pages/login.js:10]: Remove extra line.

```suggestion
```

File: pages/login.js
Comment (datderek) [inline pages/login.js:9]: Good for testing, but just don't forget to remove lines before we merge:
```
import { Amplify } from 'aws-amplify';
import awsExports from '../src/aws-exports';
Amplify.configure(awsExports);
```
These will be set up in our webapps entry point `_app.js` so the rest of the pages will have access to them.

Also I don't think `import '@aws-amplify/ui-react/styles.css';` is used?

File: pages/login.js
Comment (datderek) [inline pages/login.js:41]: Nitpick - you can just put the variable into the object. JS will automatically make the key = the variable name and the value = variable value.

```suggestion
    let user_credentials = {
      username,
      password
    }
```

File: pages/login.js
Comment (datderek) [inline pages/login.js:48]: Camelcase again.

**Review by datderek — COMMENTED** [review]
Looks good so far. I think remove the UI library for now? And just send the proper HTML response after the sign-in has been processed.

File: amplify/team-provider-info.json
Comment (sammyoto) [reply amplify/team-provider-info.json:1]: Added to gitignore but I need to delete and re-add the file to make it effective.

Comment (datderek) [reply amplify/team-provider-info.json:1]: Yeah I don't know what we are supposed to do about that, let's see what @jackfales thinks. I kept it in for now.

Comment (datderek) [inline amplify/team-provider-info.json:1]: Same as before. Let's figure this out with the captain.

File: pages/api/login.js
Comment (datderek) [inline pages/api/login.js:10]: Looks like that user variable is unused

```suggestion
    await Auth.signIn(username, password);
```

**Review by datderek — COMMENTED** [review]
Changes look good. However tried to pull your branch and run the webapp. Had this error come up when I tried to initialize the Amplify project.

image

File: pages/login.js
Comment (datderek) [inline pages/login.js:1]: Update the <input> field for the password to be `type='password'`

**Review by datderek — COMMENTED** [review]
Something is wrong with the amplify folder, don't know why I didn't have the missing 'credentials' before, but adding the config files causes conflict with 'amplify init'. I think we should wait for `[SWE-19] Create Account Backend` to merge, then you can pull the working /amplify and /src files.

**Review by datderek — APPROVED** [review]
Looks good. Works on my end - was able to login to the account I made.

**Review by datderek — APPROVED** [review]

**Pull request merged**
